File: test/graph/word-search-ii.test.ts
import { findWords } from "../../src/graph/word-search-ii";

describe('word search ii', () => {
  test('word search ii - test case 1', async () => {
    const board = [
      ['o', 'a', 'a', 'n'],
      ['e', 't', 'a', 'e'],
      ['i', 'h', 'k', 'r'],
      ['i', 'f', 'l', 'v']
    ];
    const words = ['oath', 'pea', 'eat', 'rain'];

    expect(findWords(board, words)).toMatchSnapshot();
  });

  test('word search ii - test case 2', async () => {
    const board = [
      ['a', 'b'],
      ['c', 'd']
    ];
    const words = ['abcb'];

    expect(findWords(board, words)).toStrictEqual([]);
  });

  test('word search ii - test case 3', async () => {
    const board = [['a', 'a']];
    const words = ['aaa'];

    expect(findWords(board, words)).toStrictEqual([]);
  });

  test('word search ii - test case 4', async () => {
    const board = [
      ['o', 'a', 'a', 'n'],
      ['e', 't', 'a', 'e'],
      ['i', 'h', 'k', 'r'],
      ['i', 'f', 'l', 'v']
    ];
    const words = ['oath', 'pea', 'eat', 'rain', 'hklf', 'hf'];

    expect(findWords(board, words)).toMatchSnapshot();
  });
});

File: test/leetcode/graph/word_search_ii_test.py
from leetcode.graph.word_search_ii import Solution


soln = Solution()


def test_case_1(snapshot):
    board = [["o", "a", "a", "n"], ["e", "t", "a", "e"], ["i", "h", "k", "r"], ["i", "f", "l", "v"]]
    words = ["oath", "pea", "eat", "rain"]

    result = soln.findWords(board, words)

    snapshot.assert_match(sorted(result))


def test_case_2():
    board = [["a", "b"], ["c", "d"]]
    words = ["abcb"]

    assert soln.findWords(board, words) == []


def test_case_3():
    board = [["a", "a"]]
    words = ["aaa"]

    assert soln.findWords(board, words) == []


def test_case_4(snapshot):
    board = [["o", "a", "a", "n"], ["e", "t", "a", "e"], ["i", "h", "k", "r"], ["i", "f", "l", "v"]]
    words = ["oath", "pea", "eat", "rain", "hklf", "hf"]

    result = soln.findWords(board, words)

    snapshot.assert_match(sorted(result))
